excellent/types/error.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import (
+	"fmt"
+)
+
+// XError is an error
+type XError interface {
+	XPrimitive
+	error
+}
+
+type xerror struct {
+	err error
+}
+
+// NewXError creates a new XError
+func NewXError(err error) XError {
+	return xerror{err: err}
+}
+
+// NewXErrorf creates a new XError
+func NewXErrorf(format string, a ...interface{}) XError {
+	return NewXError(fmt.Errorf(format, a...))
+}
+
+// NewXResolveError creates a new XError when a key can't be resolved on an XResolvable
+func NewXResolveError(resolvable XResolvable, key string) XError {
```

Comment:
we could create several of these factory methods, so that errors of same type are created the same way. Then at some point we could add error codes or the like to make it easier for consumers like the simulator to localize these